.gitlab-ci.yml

```diff
@@ -1,14 +1,62 @@
-test:
-  image: "python:3.6-stretch"
+stages:
+  - test
+
+.install_deps: &install_system_deps
+  before_script:
+    - >
+      dnf install -y --refresh
+      python3 python3-devel make gcc which xz libxml2-devel
+      libxslt-devel enchant python3-tox
+
+flake:
+  stage: test
+  image: fedora:latest
+  <<: *install_system_deps
   script:
-    - apt-get update -qq
-    - apt-get install -y git
-    - apt-get install -y xsltproc
-    - apt-get install -y genisoimage
-    - apt-get install -y enchant
-    - apt-get install -y shellcheck
-    - apt-get install -y python3-pip
-    - pip3 install --upgrade pip
+    - dnf install -y --refresh ShellCheck
+    - export TOXENV=check
+    - tox "-n $(nproc)"
+
+pytest:
+  stage: test
+  image: fedora:latest
+  <<: *install_system_deps
+  script:
+    - >
+      dnf install -y --refresh
+      xorriso genisoimage python36 python34 python2
+      python2-devel python2-pip
+    - export LANG=en_US.UTF-8
```

Comment:
This should not be necessary, as Fedora uses `C.UTF-8` by default.

Reply:
It is necessary. without it py2.7 fails

Reply:
Does it work with `C.UTF-8`?

Reply:
I haven't tested this, will change it and then we wait for the result

Reply:
Looks like it does.